onnxruntime/core/graph/function_utils.cc

```diff
@@ -40,20 +41,32 @@ CreateSchema(const Graph& graph,
     op_schema->TypeAndShapeInferenceFunction(meta_def->type_and_shape_inference_function);
   }
 
-  int i = 0;
+  if (allow_anytype_tensor) {
+    // if CreateSchema is used to create a schema with the 'any' type constraint,
+    // the caller (EP or otherwise) must manually check any constraints.
+    // A EP or function calls `CreateSchema` with flag allow_anytype_tensor=true, should check a node carefully
+    // whether all constraints are satisfied. Once created, this schema will match all corresponding registered kernels
+    // whenever input/output type is.
```

Comment:
suggested:

The generated schema will use the same type constraint for all inputs and outputs, and that type constraint will match all tensor types. Due to this, a user of this style of schema must manually check whether any applicable type constraints for each input or output are satisfied prior to creating a node that uses this schema.